phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/obo/go/GoGeneAnnotationParser.java

```diff
@@ -35,104 +37,57 @@
-  private final File file;
 
-  /** The {@link BufferedReader} to use for reading line-wise. */
-  private final BufferedReader reader;
 
-  /** The next line. */
-  private String nextLine;
-
-  private final List<GoGaf21Annotation> annotations;
-
-  /**
-   * Create new parser for GO gene annotation file.
-   *
-   * @param file The file to read from.
-   * @throws PhenolException In case of problems with opening and reading from <code>file</code>.
-   */
-  public GoGeneAnnotationParser(File file) throws PhenolException {
-    this.file = file;
-    try {
-      this.reader = new BufferedReader(new FileReader(file));
-      this.nextLine = reader.readLine();
-      skipHeaderAndCheckFirst();
-      ImmutableList.Builder<GoGaf21Annotation> builder = new ImmutableList.Builder<>();
-      while (hasNext()) {
-        GoGaf21Annotation annot = next();
-        builder.add(annot);
-      }
-      reader.close();
-      annotations=builder.build();
-    } catch (IOException | TermAnnotationParserException e) {
-      String msg=String.format("Could not parse GO annotation file: %s",e.getMessage());
-      throw new PhenolException(msg);
-    }
+  private GoGeneAnnotationParser() {
   }
 
-  public GoGeneAnnotationParser(String absolutepath)throws PhenolException {
-    this(new File(absolutepath));
-  }
 
-  /**
-   * Skip all header lines and check next data line.
-   *
-   * @throws TermAnnotationParserException If the first line is not as expected.
-   * @throws IOException If there is a problem with reading from the file.
-   */
-  private void skipHeaderAndCheckFirst() throws TermAnnotationParserException, IOException {
-    skipUntilData();
-    if (nextLine == null) {
-      throw new TermAnnotationParserException("GAF2.1 file contained no data!");
-    }
-    final String[] arr = nextLine.split("\t");
-    if (arr.length < 15 || arr.length > 17) {
-      throw new TermAnnotationParserException(
-          "First line of file had "
-              + arr.length
-              + " columns, but expected between 15 and 17 entries.");
-    }
+  public static List<GoGaf21Annotation> loadAnnotations(String filename) {
+    File f = new File(filename);
+    return GoGeneAnnotationParser.loadAnnotations(f);
   }
 
-  /*
-   * Skip comment lines until data
-   *
-   */
-
-  private void skipUntilData() throws IOException {
-    while (nextLine == null || nextLine.startsWith("!")) {
-      nextLine = reader.readLine();
+  public static List<GoGaf21Annotation> loadAnnotations(File file) {
+    String nextLine;
+    ImmutableList.Builder<GoGaf21Annotation> builder = new ImmutableList.Builder<>();
+    try (BufferedReader br = new  BufferedReader(new FileReader(file))) {
+      nextLine =br.readLine();
+      // skip the comments, which start with exclamation marks
+      while (nextLine.isEmpty() || nextLine.startsWith("!")) {
+        nextLine = br.readLine();
+      }
+      do {
+        final String[] arry = nextLine.split("\t");
```

Comment:
Do you need to check the array length?

Reply:
This gets checked in the constructor of the GaGofAnnotation. Perhaps it is better just to pass the line as a String and split it within the constructor? But it is checked.

Reply:
Yeah, that would be better - you could write a static GaGofAnnotation.parseAnnotation(String line) method to do all that.

Reply:
I will refactor as suggested before merging, thanks!